code/game/machinery/doors/airlock_electronics.dm

```diff
@@ -24,109 +24,118 @@
 		attack_self(user)
 		return 1
 
-	attack_self(mob/user as mob)
-		if (!ishuman(user) && !isrobot(user))
-			return ..(user)
-
-		// Can't manipulate it when broken (e.g. emagged)
-		if (icon_state == "door_electronics_smoked")
-			return
-
-		if(ishuman(user))
-			var/mob/living/carbon/human/H = user
-			if(H.getBrainLoss() >= 60)
+/obj/item/weapon/circuitboard/airlock/attackby(obj/item/W as obj, mob/user as mob)
+	if(issolder(W))
+		var/obj/item/weapon/solder/S = W
+		if(icon_state == "door_electronics_smoked")

```

Comment:
I'm not going to ask you to change this because it's existing code and probably goes scarily deep, but icon_state checks are the devil.